GsaGH/Parameters/1_Properties/GsaMaterial.cs

```diff
@@ -245,48 +246,66 @@ public enum MatType {
       IsCustom = true;
     }
 
-    internal GsaMaterial(AluminiumMaterial apiMaterial, int id) {
+    internal GsaMaterial(AluminiumMaterial apiMaterial, int id, string concreteCodeName, string steelCodeName) {
       MaterialType = MatType.Aluminium;
       _aluminiumMaterial = apiMaterial;
       _id = id;
+      ConcreteDesignCodeName = concreteCodeName;
+      SteelDesignCodeName = steelCodeName;
+      ValidateApiMaterial();
     }
 
-    internal GsaMaterial(ConcreteMaterial apiMaterial, int id, string codeName) {
+    internal GsaMaterial(ConcreteMaterial apiMaterial, int id, string concreteCodeName, string steelCodeName) {
       MaterialType = MatType.Concrete;
       _concreteMaterial = apiMaterial;
       _id = id;
-      ConcreteDesignCodeName = codeName;
+      ConcreteDesignCodeName = concreteCodeName;
+      SteelDesignCodeName = steelCodeName;
+      ValidateApiMaterial();
     }
 
-    internal GsaMaterial(FabricMaterial apiMaterial, int id) {
+    internal GsaMaterial(FabricMaterial apiMaterial, int id, string concreteCodeName, string steelCodeName) {
       MaterialType = MatType.Fabric;
       _fabricMaterial = apiMaterial;
       _id = id;
+      ConcreteDesignCodeName = concreteCodeName;
```

Comment:
can you move this part into separate method and call it?

Reply:
done